test/verify/check-client

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/python3 -cimport os, sys; os.execv(os.path.dirname(sys.argv[1]) + "/../common/pywrap", sys.argv)
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import testlib
+
+
+# Run this on more OSes as we roll out the pybridge
+@testlib.onlyImage("needs pybridge", "debian-testing")
+class TestClient(testlib.MachineCase):
+
+    provision = {
+        "client": {"address": "10.111.113.1/24", "memory_mb": 660},
+        "target": {"address": "10.111.113.2/24", "memory_mb": 660},
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.machine.upload(["../src/client/cockpit-beiboot"], "/usr/local/bin/")
+        self.machines["target"].execute("hostnamectl set-hostname target")
+        # FIXME: create cache dir in the client
+        self.machine.execute("runuser -u admin -- mkdir -p ~admin/.cache")
+
+    def writeConfig(self, ssh_command):
+        self.machine.write("/etc/cockpit/cockpit.conf", f"""
+[WebService]
+X-For-CockpitClient = true
+LoginTo = true
+
+[Ssh-Login]
+ReportStderr = true
+Command = {ssh_command}
+""")
+
+    def logout(self, check_last_host=None):
+        b = self.browser
+
+        b.assert_no_oops()
+        b.open_session_menu()
+        b.click('#logout')
+        # FIXME: This is broken, nothing appears
+        # b.wait_text("#brand", "Connect to:")
+        if check_last_host:
+            b.wait_val("#server-field", check_last_host)
+
+        # no leaked processes
+        # FIXME: leaks ssh on debian-testing with c-client-ssh (but not on fedora-38/pybridge)
+        self.machine.execute('''
+            while [ -n "$(pgrep -au admin | grep -Ev 'cockpit-ws|ssh.*cockpit-bridge' >&2)" ]; do sleep 1; done
+        ''', timeout=10)
+        # FIXME: ssh-agent leaks both with and without remote bridge; fixed in PR#18917
+        self.machines["target"].execute("while pgrep -af '([c]ockpit|[s]sh-agentFIXME)' >&2; do sleep 1; done",
```

Comment:
You don't need the `[w]eird|[t]ricks` here — pgrep never matches itself.

Reply:
Oh.  But it would match the shell which is executing it.  I guess we pass the command via `-c` and not via `stdin`?.

Reply:
Yes, it always caught itself. I don't "automatically" write this, I needed the failing tests to tell me :)